src/frontend/Semantic_error.ml

```diff
@@ -352,6 +353,9 @@ module StatementError = struct
         Fmt.pf ppf
           "Cannot assign to global variable '%s' declared in previous blocks."
           name
+    | LValueMultiIndexing ->
+        Fmt.pf ppf
+          "Left hand side of an assignment cannot have nested multi-indexing."
```

Comment:
More of a question than anything - after removal do you think we should change the parser/AST to enforce this, or keep it in typechecking? 

Is the eventual hope to re-introduce with matching semantics?

Reply:
I don't know. Cleaning up the AST seems like a good idea but nested indexing is still allowed when all the indexes are simple integers. If we change the AST then pretty printer will always flatten the indexes. That may be acceptable, it's not like pretty printer preserves all the formatting anyway.
In my opinion nested indexing is just an accidental consequence of the ability to index any expression. It's not necessary to support it in lvalue position, where arbitrary expressions aren't allowed.

Reply:
Fair enough. Does this change do anything for nested indexing on the left hand side of a `~`, or is that technically fine since it ends up inside a `lpdf`? I'm fuzzy on whether that counts as an l-value or not.

Reply:
No change there, the "left-hand side" of a tilde statement is not an lvalue.